### ISSUE TITLE
Question: for old-ass Cary 50 BKN data

### ISSUE THREAD

**Issue by idikiy** [mindcruzer/bkn-reader#1]
Hello,  I tried this reader on some really old Kinetics data from a Cary 50 spectrophotometer (yes we still use one of those) and it's not enjoying that version of BKN format, giving me a segmentation fault.  Since I know very little/almost nothing about programming, I was wondering if this looks like an issue I could fix on my own with some walking-through, or if I should just give up and get some newer software.  I think the error arises from a different way the "Parameter List" is coded in the older BKN files.  A sample old BKN file is attached.
[ID120s.BKN.zip](https://github.com/mindcruzer/bkn-reader/files/3695248/ID120s.BKN.zip)

**Comment by mindcruzer** [comment]
I have to admit I never thought anyone would actually use this.    

You're probably much better off getting some newer software. I threw this together like 8 years ago on a whim, and I seem to recall spending quite a bit of time with a hex editor researching the structure of the files. If your format is a bit different then you'd need to get a hex editor and check out the differences between your file and the sample file to see where the code needs modifying. Could be a fun exercise to get familiar with C and reading data from files, but if you don't have any programming experience it will probably just make you want to smash your head against a wall.    

I'd do it, and it does sound kind of fun, but unfortunately I'm buried in work right now ಥ╭╮ಥ    

I'll probably come back to this and check it out eventually but likely won't be able to for a month or so.

**Comment by idikiy** [comment]
Yesss I got it to work!

Thanks for the pointer to using a hex editor - basically the offset was a bit different in one spot, so it was reading the rest of the file into memory and crashing.  Also the fields are a bit different between the two versions, so I just ignored them.  I didn't really get a working JSON file out of it but I got my data out of a corrupted file, which is all I cared about.

I was basically just editing the read_bkn.c file and re-making on my machine.  What's the best way for me make it into a different version, so that I can have both your program and this jury-rigged one?  I might use one or the other in the future (unlikely, but still).
